### PR TITLE
Very .m2/.gradle-enterprise is not polluted

### PR DESCRIPTION
**Pull request by tresat**
Add enterprise access key to build so that scans get published on CI.

Check that the .m2/gradle-enterprise directory is clear and also parameterize all the directory empty checks.

### CODE REVIEW

File: .teamcity/src/main/kotlin/promotion/PromotionProject.kt [hunk 1]
```diff
@@ -39,6 +39,7 @@ class PromotionProject(branch: VersionedSettingsBranch) : Project({
         param("env.ORG_GRADLE_PROJECT_sdkmanKey", "8ed1a771bc236c287ad93c699bfdd2d7")
         param("env.PGP_SIGNING_KEY", "%pgpSigningKey%")
         param("env.PGP_SIGNING_KEY_PASSPHRASE", "%pgpSigningPassphrase%")
+        password("env.GRADLE_ENTERPRISE_ACCESS_KEY", "%ge.gradle.org.access.key%;%e.grdev.net.access.key%")
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/promotion/PromotionProject.kt:42]: We don't need this here, this is already set on the individual build configurations, right? See `BasePromotionBuildType`.

**Review by wolfs — COMMENTED** [review]
Looks good in general, I left some smaller comments.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**